cupy/cuda/cupy_memory.h

```diff
@@ -0,0 +1,63 @@
+#ifndef INCLUDE_GUARD_CUPY_CUDA_MEMORY_H
+#define INCLUDE_GUARD_CUPY_CUDA_MEMORY_H
+
+#include "memory_api.h"  // generated by Cython from cupy.cuda.memory
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+  *********************** C public API ************************
+*/
+typedef struct cupy_allocator_handle_t cupy_allocator_handle;
```

Comment:
Should the header be documented?

Reply:
Oh yes, see 5a4aadc. Pending @emcastillo's feedback on the style and appearance...